Massive amount of failures concerning async in Jakarta EE 8 TCK samples

Nearly evert test in https://github.com/javaee-samples/jakartaee8-tck that passed before now throws:

```
javax.servlet.ServletException: java.lang.IllegalStateException: Async is not supported
	at org.jakartaee8.servlet.servletrequest.asynccontext.GenericTCKServlet.invokeTest(GenericTCKServlet.java:88)
	at org.jakartaee8.servlet.servletrequest.asynccontext.GenericTCKServlet.service(GenericTCKServlet.java:72)
	at cloud.piranha.webapp.impl.DefaultFilterChain.doFilter(DefaultFilterChain.java:114)
	at cloud.piranha.security.exousia.AuthorizationFilter.doFilter(AuthorizationFilter.java:86)
	at cloud.piranha.security.exousia.AuthorizationFilter.doFilter(AuthorizationFilter.java:76)
	at cloud.piranha.webapp.impl.DefaultFilterChain.doFilter(DefaultFilterChain.java:110)
	at cloud.piranha.security.eleos.AuthenticationFilter.doFilter(AuthenticationFilter.java:82)
	at cloud.piranha.security.eleos.AuthenticationFilter.doFilter(AuthenticationFilter.java:76)
	at cloud.piranha.webapp.impl.DefaultFilterChain.doFilter(DefaultFilterChain.java:110)
	at cloud.piranha.security.exousia.AuthorizationPreFilter.doFilter(AuthorizationPreFilter.java:94)
	at cloud.piranha.security.exousia.AuthorizationPreFilter.doFilter(AuthorizationPreFilter.java:79)
	at cloud.piranha.webapp.impl.DefaultFilterChain.doFilter(DefaultFilterChain.java:110)
	at cloud.piranha.webapp.impl.DefaultServletRequestDispatcher.request(DefaultServletRequestDispatcher.java:135)
	at cloud.piranha.webapp.impl.DefaultWebApplication.service(DefaultWebApplication.java:1573)
	at cloud.piranha.appserver.impl.DefaultWebApplicationServer.service(DefaultWebApplicationServer.java:285)
	at cloud.piranha.appserver.impl.DefaultWebApplicationServer.process(DefaultWebApplicationServer.java:244)
	at cloud.piranha.http.impl.DefaultHttpServerProcessingThread.run(DefaultHttpServerProcessingThread.java:85)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.lang.IllegalStateException: Async is not supported
	at cloud.piranha.webapp.impl.DefaultWebApplicationRequest.startAsync(DefaultWebApplicationRequest.java:1586)
	at org.jakartaee8.servlet.servletrequest.asynccontext.AsyncTestServlet.dispatchContextPathTest1(AsyncTestServlet.java:114)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.jakartaee8.servlet.servletrequest.asynccontext.GenericTCKServlet.invokeTest(GenericTCKServlet.java:86)
	... 19 more
javax.servlet.ServletException: java.lang.IllegalStateException: Async is not supported
	at org.jakartaee8.servlet.servletrequest.asynccontext.GenericTCKServlet.invokeTest(GenericTCKServlet.java:88)
	at org.jakartaee8.servlet.servletrequest.asynccontext.GenericTCKServlet.service(GenericTCKServlet.java:72)
	at cloud.piranha.webapp.impl.DefaultFilterChain.doFilter(DefaultFilterChain.java:114)
	at cloud.piranha.security.exousia.AuthorizationFilter.doFilter(AuthorizationFilter.java:86)
	at cloud.piranha.security.exousia.AuthorizationFilter.doFilter(AuthorizationFilter.java:76)
	at cloud.piranha.webapp.impl.DefaultFilterChain.doFilter(DefaultFilterChain.java:110)
	at cloud.piranha.security.eleos.AuthenticationFilter.doFilter(AuthenticationFilter.java:82)
	at cloud.piranha.security.eleos.AuthenticationFilter.doFilter(AuthenticationFilter.java:76)
	at cloud.piranha.webapp.impl.DefaultFilterChain.doFilter(DefaultFilterChain.java:110)
	at cloud.piranha.security.exousia.AuthorizationPreFilter.doFilter(AuthorizationPreFilter.java:94)
	at cloud.piranha.security.exousia.AuthorizationPreFilter.doFilter(AuthorizationPreFilter.java:79)
	at cloud.piranha.webapp.impl.DefaultFilterChain.doFilter(DefaultFilterChain.java:110)
	at cloud.piranha.webapp.impl.DefaultServletRequestDispatcher.request(DefaultServletRequestDispatcher.java:135)
	at cloud.piranha.webapp.impl.DefaultWebApplication.service(DefaultWebApplication.java:1573)
	at cloud.piranha.appserver.impl.DefaultWebApplicationServer.service(DefaultWebApplicationServer.java:285)
	at cloud.piranha.appserver.impl.DefaultWebApplicationServer.process(DefaultWebApplicationServer.java:244)
	at cloud.piranha.http.impl.DefaultHttpServerProcessingThread.run(DefaultHttpServerProcessingThread.java:85)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.lang.IllegalStateException: Async is not supported
	at cloud.piranha.webapp.impl.DefaultWebApplicationRequest.startAsync(DefaultWebApplicationRequest.java:1586)
	at org.jakartaee8.servlet.servletrequest.asynccontext.AsyncTestServlet.dispatchContextPathTest1(AsyncTestServlet.java:114)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.jakartaee8.servlet.servletrequest.asynccontext.GenericTCKServlet.invokeTest(GenericTCKServlet.java:86)
	... 19 more
```

--- a/security/eleos/src/main/java/cloud/piranha/security/eleos/AuthenticationInitializer.java
+++ b/security/eleos/src/main/java/cloud/piranha/security/eleos/AuthenticationInitializer.java
@@ -34,6 +34,7 @@ import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
 
+import javax.servlet.FilterRegistration;
 import javax.servlet.ServletContainerInitializer;
 import javax.servlet.ServletContext;
 import javax.servlet.ServletException;
@@ -91,9 +92,11 @@ public class AuthenticationInitializer implements ServletContainerInitializer {
         DefaultAuthenticationService authenticationService = new DefaultAuthenticationService(appContextId, options, configParser, null);
         
         servletContext.setAttribute(AUTH_SERVICE, authenticationService);
-        
-        servletContext.addFilter(AuthenticationFilter.class.getSimpleName(), AuthenticationFilter.class);
-        
+
+        FilterRegistration.Dynamic dynamic = servletContext.addFilter(AuthenticationFilter.class.getSimpleName(), AuthenticationFilter.class);
+
+        dynamic.setAsyncSupported(true);
+
         ((WebApplication) servletContext).addFilterMapping(AuthenticationFilter.class.getSimpleName(), "/*");
     }
 

--- a/security/exousia/src/main/java/cloud/piranha/security/exousia/AuthorizationInitializer.java
+++ b/security/exousia/src/main/java/cloud/piranha/security/exousia/AuthorizationInitializer.java
@@ -29,6 +29,7 @@ package cloud.piranha.security.exousia;
 
 import java.util.Set;
 
+import javax.servlet.FilterRegistration;
 import javax.servlet.ServletContainerInitializer;
 import javax.servlet.ServletContext;
 import javax.servlet.ServletException;
@@ -54,7 +55,9 @@ public class AuthorizationInitializer implements ServletContainerInitializer {
 
         WebApplication context = (WebApplication) servletContext;
 
-        servletContext.addFilter(AuthorizationFilter.class.getSimpleName(), AuthorizationFilter.class);
+        FilterRegistration.Dynamic dynamic = servletContext.addFilter(AuthorizationFilter.class.getSimpleName(), AuthorizationFilter.class);
+
+        dynamic.setAsyncSupported(true);
 
         context.addFilterMapping(AuthorizationFilter.class.getSimpleName(), "/*");
     }

--- a/security/exousia/src/main/java/cloud/piranha/security/exousia/AuthorizationPreInitializer.java
+++ b/security/exousia/src/main/java/cloud/piranha/security/exousia/AuthorizationPreInitializer.java
@@ -40,6 +40,7 @@ import java.util.Set;
 
 import javax.security.jacc.PolicyConfiguration;
 import javax.security.jacc.PolicyContextException;
+import javax.servlet.FilterRegistration;
 import javax.servlet.ServletContainerInitializer;
 import javax.servlet.ServletContext;
 import javax.servlet.ServletException;
@@ -121,8 +122,8 @@ public class AuthorizationPreInitializer implements ServletContainerInitializer 
         }
 
         servletContext.setAttribute(AUTHZ_SERVICE, authorizationService);
-        servletContext.addFilter(AuthorizationPreFilter.class.getSimpleName(), AuthorizationPreFilter.class);
-
+        FilterRegistration.Dynamic dynamic = servletContext.addFilter(AuthorizationPreFilter.class.getSimpleName(), AuthorizationPreFilter.class);
+        dynamic.setAsyncSupported(true);
         context.addFilterMapping(AuthorizationPreFilter.class.getSimpleName(), "/*");
     }
 

